certbot-nginx/certbot_nginx/configurator.py

```diff
@@ -162,6 +162,14 @@ def prepare(self):
         if self.version is None:
             self.version = self.get_version()
 
+        # Prevent two Nginx plugins from modifying a config at once
+        try:
+            util.lock_dir_until_exit(self.conf('server-root'))
```

Comment:
One dayt, we'd probably want to make this a part of `IAuthenticator` and `IInstaller`, so that third party plugin authors don't have to know to make locking calls themselves (ie the API can suggest they should provide a directory to lock).